pkg/scheduler/framework/plugins/dynamicresources/structuredparameters_test.go

```diff
@@ -0,0 +1,1259 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamicresources
+
+import (
+	"errors"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	v1 "k8s.io/api/core/v1"
+	resourceapi "k8s.io/api/resource/v1alpha2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	namedresourcesmodel "k8s.io/kubernetes/pkg/scheduler/framework/plugins/dynamicresources/structured/namedresources"
+	"k8s.io/kubernetes/test/utils/ktesting"
+	"k8s.io/utils/ptr"
+)
+
+func TestModel(t *testing.T) {
+	testcases := map[string]struct {
+		slices   resourceSliceLister
+		claims   assumeCacheLister
+		inFlight map[types.UID]resourceapi.ResourceClaimStatus
+
+		expectResources resources
+		expectErr       string
+	}{
+		"empty": {},
+
+		"slice-list-error": {
+			slices: sliceError("slice list error"),
+
+			expectErr: "list node resource slices: slice list error",
+		},
+
+		"unknown-model": {
+			slices: sliceList{
+				&resourceapi.ResourceSlice{
+					NodeName:      "node",
+					DriverName:    "driver",
+					ResourceModel: resourceapi.ResourceModel{ /* empty! */ },
+				},
+			},
+
+			// Not an error. It is safe to ignore unknown resources until a claim requests them.
+			// The unknown model in that claim then triggers an error for that claim.
+			expectResources: resources{"node": map[string]ResourceModels{
+				"driver": {
+					NamedResources: namedresourcesmodel.Model{},
+				},
+			}},
+		},
+
+		"one": {
```

Comment:
Avoiding some variables is okay because their values are short, but repeating for example `twoInstances` whenever it's used IMHO would make it hard to see what the difference between two test cases is.

Reply:
I thought you were still referring to `namedresourcesmodel_test.go` because that is where I define common values in variables. Here I don't, but perhaps I should? It's not clear whether you call this out as a good example for tests or want me to change something :sweat_smile: 

"one" -> "one-instance"